op-batcher/batcher/channel_manager.go

```diff
@@ -144,6 +175,42 @@ func (s *channelManager) nextTxData(channel *channel) (txData, error) {
 	return tx, nil
 }
 
+func (s *channelManager) HasTxData() bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, ch := range s.channelQueue {
+		if ch.HasTxData() {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *channelManager) ProcessBlocks(l1Head eth.BlockID) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// If we have no saved blocks, we will not be able to create valid frames
+	if len(s.blocks) == 0 {
+		return io.EOF
```

Comment:
should this be `io.EOF` ? or be a custom error?

Reply:
`io.EOF` is already used in many places in the codebase, so we're just following that pattern.
Generally io.EOF is used to signify end of a read stream. the channelManager is not technically a read stream, but I think returning io.EOF is generally acceptable. Otherwise we'd just define our own custom error which would just do the same thing.